js/coinex.js

```diff
@@ -990,6 +1030,100 @@ module.exports = class coinex extends Exchange {
         return this.parseTrades (trades, market, since, limit);
     }
 
+    async transfer (code, amount, fromAccount, toAccount, params = {}) {
+        await this.loadMarkets ();
+        const currency = this.currency (code);
+        const accountsByType = this.safeValue (this.options, 'accountsByType', {});
+        const fromId = this.safeNumber (accountsByType, fromAccount);
+        const toId = this.safeNumber (accountsByType, toAccount);
+        const request = {
+            'coin_type': currency['id'],
+            'amount': this.currencyToPrecision (code, amount),
+        };
+        let method = undefined;
+        if (fromAccount === 'margin' || toAccount === 'margin') {
+            method = 'privatePostMarginTransfer';
+            const symbol = this.safeString (params, 'symbol');
+            if (symbol === undefined) {
+                throw new ExchangeError ('You must define in params what market you are transferring. e.g. {symbol: ETH/BTC}');
+            }
+            const market = this.market (symbol);
+            request['market'] = market['id'];
+            const marginMarkets = await this.publicGetMarginMarket (params);
```

Comment:
I'm not sure if we should do this `publicGetMarginMarket()` request on each call to `transfer()`. The rule of thumb is to have one HTTP request per unified method, with very few exceptions. In this case, I think  `publicGetMarginMarket()` belongs to `fetchMarkets()`, where we can issue more than one request, loading and caching them on the fly. So, I suggest that we move this call to `fetchMarkets` once and reuse this info on consequent calls to `transfer()`.

For example: 

- https://github.com/ccxt/ccxt/blob/master/js/gateio.js#L640

Reply:
Sounds good! I thought the same, I just wasn't sure if it was ok to add `accountId` to the market object. I'll make the change